agg-ast/schema_derivation/src/schema_derivation_tests/stage.rs

```diff
@@ -1268,31 +1268,187 @@ mod project {
             ..Default::default()
         })
     );
-    // SQL-2785
-    // test_derive_stage_schema!(
-    //     project_array_of_docs,
-    //     expected = Ok(Schema::Document(Document {
-    //         keys: map! {
-    //             "foo".to_string() => Schema::Array(Box::new(Schema::Document(Document {
-    //                 keys: map! {
-    //                     "a".to_string() => Schema::Atomic(Atomic::String),
-    //                     "b".to_string() => Schema::Atomic(Atomic::Integer),
-    //                 },
-    //                 ..Default::default()
-    //             })))
-    //         },
-    //         required: set!("foo".to_string()),
-    //         ..Default::default()
-    //     )),
-    //     input = r#"{"$project": {"foo.a": 1, "foo.b": 1}}"#,
-    //     ref_schema = Schema::Array(Box::new(Schema::Document(Document {
-    //         keys: map! {
-    //             "a".to_string() => Schema::Atomic(Atomic::String),
-    //             "b".to_string() => Schema::Atomic(Atomic::Integer),
-    //         },
-    //         ..Default::default()
-    //     })))
-    // );
+    test_derive_stage_schema!(
+        project_multiple_fields_array_of_docs,
```

Comment:
Can you also a add a test very similar to this one but with extra keys in the `ref_schema` that are not projected in the `input`. As in, what if the `ref_schema` had a `c` field and the `input` stayed the same as it is? The output should be the same as well in that case. We should have a test to assert this since it could be possible for us to mess up the implementation where we retain all fields.